src/shims/foreign_items/windows.rs

```diff
@@ -213,6 +213,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 // (Windows locks are reentrant, and we have only 1 thread,
                 // so not doing any futher checks here is at least not incorrect.)
             }
+            "TryEnterCriticalSection" if this.frame().instance.to_string().starts_with("std::sys::windows::")
+            => {
+                // There is only one thread, so this always succeeds and returns TRUE
+                this.write_scalar(Scalar::from_int(1, Size::from_bytes(4)), dest)?;
```

Comment:
`Scalar::from_i32` is more convenient here than `Scalar::from_int`.

Reply:
I get an assertion failure on 1 != 4 when using `Size::from_bytes(4)` or `from_i32()`, but it works when using `dest.layout.size`. Something fishy is going on here, I'll have to look into this further.

Reply:
That probably means the return type actually has size 1, so it should be `from_i8` (or `from_u8`).

Reply:
But according to the docs, the return type is `BOOL`, which on Windows should be `i32`...

Reply:
Fun fact: the libstd prototype disagrees with the one in the Windows docs. I'll report a libstd bug.

This is exactly why I prefer `from_<fixed tye>` over using `dest.layout.size`. ;)

Reply:
Reported as https://github.com/rust-lang/rust/issues/70504.

I'd say you should use `from_u8` here and add a FIXME linking to that issue.